### ISSUE TITLE
Drawer incompatible with `react-native-reanimated@3.2.0` on android but compatible with `v3.1.0`

### ISSUE THREAD

**Issue by paulrostorp** [react-community/react-navigation#11391]
### Current behavior

Displaying the drawer (via `dispatch(DrawerActions.toggleDrawer)`) doesn't work with `react-native-reanimated@3.2.0`.

https://github.com/react-navigation/react-navigation/assets/1866649/d9011474-6b87-4037-b0ac-dea4ff3f97cb

As you can see, the status bar changes but the drawer doesn't display

### Expected behavior

It should work :)
I haven't digged into to why it breaks, but **downgrading to RNR 3.1.0** fixes the issues.
The release is pretty big: https://github.com/software-mansion/react-native-reanimated/releases/tag/3.2.0

### Reproduction

not yet

### Platform

- [X] Android
- [x] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [X] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [ ] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [x] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | 6.1.6
| @react-navigation/drawer               | 6.6.2
| @react-navigation/stack                | 6.3.16 
| @react-navigation/native-stack         |  6.9.7
| react-native-safe-area-context         | 4.5.3
| react-native-screens                   | 3.20.0
| react-native-gesture-handler           | 2.10.1
| react-native-reanimated                | 3.2.0
| react-native                           | 0.71.8
| node                                   | 16.14.2
| npm or yarn                            | yarn 3.3.0

**Comment by gavrichards** [comment]
On iOS, the whole screen just goes white when you tap to open the drawer.

**Comment by ycdaskin** [comment]
same with ios. drawer never opens after closing it once. but the state of drawer changes. i am swiping from left of the screen to the right. drawer doesn't appear but its status changes. i can confirm it with the code below.

```
const isOpen = useDrawerStatus() === 'open'

useEffect(() => {
        if (isOpen) {
            console.log("drawer open")
        } else {
            console.log("drawer closed")
        }
    }, [isOpen])
```

**Comment by pavandugariya** [comment]
> The versions mentioned in the issue for the following packages differ from the latest versions on npm:
> 
> * `@react-navigation/native` (found: `6.1.1`, latest: `6.1.6`)
> * `@react-navigation/drawer` (found: `6.5.6`, latest: `6.6.2`)
> * `@react-navigation/stack` (found: `6.3.10`, latest: `6.3.16`)
> 
> Can you verify that the issue still exists after upgrading to the latest versions of these packages?
I have used 

  "dependencies": {
    "@react-navigation/drawer": "^6.6.2",
    "@react-navigation/material-bottom-tabs": "^6.2.15",
    "@react-navigation/native": "^6.1.6",
    "@react-navigation/native-stack": "^6.3.16",
    "react": "18.2.0",
    "react-native": "0.71.8",
    "react-native-gesture-handler": "~2.8.0",
    "react-native-linear-gradient": "^2.6.2",
    "react-native-modal": "^13.0.1",
    "react-native-paper": "^5.8.0",
    "react-native-reanimated": "^3.1.0",
    "react-native-safe-area-context": "^4.4.1",
    "react-native-screens": "~3.18.0",
    "react-native-vector-icons": "^9.2.0"
  },

but still not working ..

**Comment by pipo151086** [comment]
same here.......

**Comment by imomer** [comment]
I'm having the same issue but mine doesn't work even following their official guide: https://reactnavigation.org/docs/drawer-navigator#installation

In the guide, they said that drawer navigation supports reanimated v1.X and v2.X but now there is v3.X

**Comment by YuBeen-Park** [comment]
I'm having the same issue.
It worked when the version was "react-native-reanimated": "^3.0.2", but suddenly it's not working even on version 3.1.0 and 3.2.0
I've checked the useDrawerStatus(), drawer state was changing normally but the drawer doesn't display.

**Comment by AlirezaHadjar** [comment]
https://github.com/react-navigation/react-navigation/assets/57192409/454596a4-d10e-496f-b851-2082d86d8194

I haven't had any issues with v3.1.0 but apparently, v3.2.0 isn't compatible with drawer.

**Comment by FernandoAOborges** [comment]
Same here. Any news?

**Comment by ivan-zeller** [comment]
Same for version 3.2.0

**Comment by gavrichards** [comment]
Any chance we could stop with the "same here" comments please? I've subscribed to this thread, to be notified when there's a fix. I don't see how it's helpful.

**Comment by ryanalencar** [comment]
@YuBeen-Park I tried version 3.0.2 and does not work either.

On my app, I'm using the following versions:
"react-native-reanimated": "^3.0.2",
"react-native-safe-area-context": "^4.5.3",
"react-native-screens": "^3.20.0",
"@react-navigation/drawer": "^6.6.2",
"@react-navigation/native": "^6.1.6",
"@react-navigation/native-stack": "^6.9.12",

**Comment by worldpeacez0991** [comment]
#Edited android\gradle.properties
newArchEnabled=true

[react-native-reanimated] Reanimated 3.2.0 supports the New Architecture only on the latest minor release of React Native. Please upgrade to React Native 0.72.0+ or downgrade to an older version of Reanimated v3.

I tried to migrate from "react-native": "0.71.8" to "0.72.0-rc.5"
Went through all the steps to bring up my app nicely.
However, when I click on the drawer, drawer is not showing. (i.e. 0.72.0-rc.5 does not work with Reanimated v3)

package.json:

 ```
"dependencies": {
    "react": "18.2.0",
    "react-native": "0.72.0-rc.5",
    "react-native-reanimated": "^3.2.0",
    "react-native-safe-area-context": "^4.5.3",
    "react-native-screens": "^3.20.0",
    "@react-navigation/drawer": "^6.6.2",
    "@react-navigation/native": "^6.1.6",
    "@react-navigation/stack": "^6.3.16",
    "react-native-gesture-handler": "^2.10.2"
 }
```

Hope this saves someone time, to not try this path
and... maybe, will try to downgrade Reanimated to v2

**Comment by AbdelmadjidBOUR** [comment]
I had the same issue and I just solved it now. I was using RNR "^3.1.0", I tried to upgrade it to "^3.2.0" but the issue persisted, I downgraded again to "^3.1.0" and I cleared the cash using : "npm start -- --reset-cache", rebuilt the app. The problem solved.

**Comment by FernandoAOborges** [comment]
I solved using  "react-native-reanimated": "3.1.0".

**Comment by ivan-zeller** [comment]
But jest tests run failed on "react-native-reanimated": "3.1.0" if i use react-native-gesture handler of any version

**Comment by abdurrehmanoffice** [comment]
Exactly 10 days before when using the same code the drawer were opening with swipe and onPress function both. 
But now 
       initially drawer opens by swiping and not by onpress.
       even by onpress function drawer by swiping does not work 

onpress={() => this.props.navigation.openDrawer()}


"dependencies": {
    "@react-navigation/bottom-tabs": "^6.5.7",
    "@react-navigation/drawer": "^6.6.2",
    "@react-navigation/native": "^6.1.6",
    "@react-navigation/native-stack": "^6.9.12",
    "react": "18.2.0",
    "react-native": "0.71.8",
    "react-native-gesture-handler": "^2.10.1",
    "react-native-reanimated": "^3.1.0",
    "react-native-safe-area-context": "^4.5.3",
    "react-native-screens": "^3.20.0",
  },

**Comment by C0d3Br3aker** [comment]
I solved it by using "react-native-reanimated": "3.3.0".

**Comment by larsmunkholm** [comment]
> I solved it by using "react-native-reanimated": "3.3.0".

The only change seems to be updating react-native-gesture-handler to ^2.11.0.

I'm gonna see if this solves my issue later today.

**Comment by worldpeacez0991** [comment]
Cool, drawer navigation works!
react-native-reanimated": "^3.3.0" is compatible with "react-native": "0.71.10". -- PASS

package.json
```
"dependencies": {
    "@react-navigation/drawer": "^6.6.2",
    "@react-navigation/native": "^6.1.6",
    "react": "18.2.0",
    "react-native": "0.71.10",
    "react-native-gesture-handler": "^2.12.0",
    "react-native-reanimated": "^3.3.0",
    "react-native-safe-area-context": "^4.5.3",
    "react-native-screens": "^3.20.0"
  },

```  
Feel free to check this sample project : https://github.com/worldpeacez0991/rnDrawerNavigation

**Comment by edstbbz** [comment]
> Cool, drawer navigation works! react-native-reanimated": "^3.3.0" is compatible with "react-native": "0.71.10". -- PASS
> 
> package.json
> 
> ```
> "dependencies": {
>     "@react-navigation/drawer": "^6.6.2",
>     "@react-navigation/native": "^6.1.6",
>     "react": "18.2.0",
>     "react-native": "0.71.10",
>     "react-native-gesture-handler": "^2.12.0",
>     "react-native-reanimated": "^3.3.0",
>     "react-native-safe-area-context": "^4.5.3",
>     "react-native-screens": "^3.20.0"
>   },
> ```
> 
> Feel free to check this sample project : https://github.com/worldpeacez0991/rnDrawerNavigation

This doesn't work if the Chrome debugger is enabled. When the debugger is enabled, it simply does not open

**Comment by edstbbz** [comment]
> have you check if you are using the correct version in your file 'yarn.lock'? try these steps Via package.json > remove line 'react-native-reanimated@^3.3.0' $ yarn $ yarn add react-native-reanimated@^3.3.0
> 
> And, i only tested for android mobile, not sure how Chrome debugger is different. anyone, please help, thanks

Yes, if not connected to Chrome debugger it worked before, but if you use a debugger, drawer does not open.
Another problem, after update react-native > 0.67 there is a short suspension when switching from drawer

**Comment by worldpeacez0991** [comment]
"if you use a debugger, drawer does not open."
Care to reproduce issue by;
Sharing minimal code in github, with steps and GIF in README to reproduce issue, please

**Comment by edstbbz** [comment]
> "if you use a debugger, drawer does not open." Care to reproduce issue by; Sharing minimal code in github, with steps and GIF in README to reproduce issue, please

https://github.com/worldpeacez0991/rnDrawerNavigation - Reproduced here

**Comment by worldpeacez0991** [comment]
This project GIF looks working.
Can you reconfirm you share the correct link?
I wish i could help you, could you share steps or GIF for your screen, please

**Comment by worldpeacez0991** [comment]
I retested [rnDrawerNavigation](https://github.com/worldpeacez0991/rnDrawerNavigation), seems to have no issue with Android.
I am currently missing the test steps using Chrome Debugger, if you are free to share the steps, i will be glad to help you

**Comment by edstbbz** [comment]
> Я повторно протестировал [rnDrawerNavigation](https://github.com/worldpeacez0991/rnDrawerNavigation) , вероятно, у него нет проблем с Android. В настоящее время мне не хватает тестовых шагов с помощью Chrome Debugger, если вы можете поделиться шагами, я буду рад вам помочь

You just need to launch your project, connect to Chrome debugging and you will see that it stops opening

**Comment by worldpeacez0991** [comment]
i see..I am quite new to webview.

After some research, i realise to inspect with Chrome debugger, your project is using something like this in your package.json:
"react-native-webview": "^13.2.2"

Currently, i can see from:
chrome://inspect/#devices
![WebView_snap1](https://github.com/react-navigation/react-navigation/assets/36908871/05bd80db-0d7a-485a-b8c8-b6f473d87d33)

My App.tsx has no logic for drawer navigation yet.
I did tested, to see that if i link to a URL (which already has drawer navigation, this will work as well)
I clicked on drawer icon on top left, then clicked on 'showcase'
https://github.com/react-navigation/react-navigation/assets/36908871/b2a093fc-dffc-4320-9b7d-75e442607960

Now, let me try if i can implement drawer navigation without this URL... will update if i have any success

**Comment by TechSaq** [comment]
> 

This worked for me

**Comment by Vahabov** [comment]
How you solved this?.. Still not working :(

**Comment by garrettmac** [comment]
yeah please prioritize this bug. If someone has a solve with it working with the Chrome debugger please share

**Comment by da8ah** [comment]
# Expo

I had the same issue with Expo Router v2 (Drawer), so this worked for me.

https://github.com/software-mansion/react-native-reanimated/issues/3796#issuecomment-1793609970

1. `rm -rf node_modules/`
2. Verify dependencies and reinstall them `npm i`.
3. Add a `resolution` section in `package.json` with reanimated@3.3.0 and reinstall `npm i`.
4. The reanimated plugin must be the last item in the plugin array at `babel.config.js`.
5. Set `hermes` engine, [Reanimated 1 work with Chrome debugger. Version 2+ needs hermes](https://reactnavigation.org/docs/drawer-navigator/#uselegacyimplementation).
6. Start with clean cache `npx expo start -c`.

I hope this helps!

**Comment by pollyolly** [comment]
I use react-native-reanimated@2.2 works on the latest

**Comment by eidan66** [comment]
```
    "@react-navigation/drawer": "^6.6.6",
    "react-native-reanimated": "^3.6.1",
    "react-native": "0.72.6",
```

Same error here, and downgrade didn't help

**Comment by djbuch** [comment]
I think it coul be corrected with : https://github.com/react-navigation/react-navigation/pull/11796 but the release isn't out yet

**Comment by izavyalov-dev** [comment]
Works for me with following configuration, looks like in 7.x version this is fixed.

```
    "@react-navigation/drawer": "^7.0.0-alpha.21",
    "@react-navigation/native": "^7.0.0-alpha.18",
    "@react-navigation/native-stack": "^7.0.0-alpha.20",
    "@react-navigation/stack": "^7.0.0-alpha.20",
    "react-native-reanimated": "^3.11.0",
```
